File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    int sum = 0;
    
    for(int i = 1; i<=n; i++){
        sum += i;
        if(sum >= n){
            answer++;
        }
        for(int j = i+1; j<=n; j++){
            sum += j;
            if(sum == n){
                answer++;
                break;
            }
            else if(sum > n) break;
        }
        
        sum = 0;
    }
    
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        
        int left = 1;
        int right = 1;
        int sum = 0;
        
        while(n >= right) {
            if(sum == n) answer++;

            if(sum >= n) sum -= left++;

            else if(sum < n) sum += right++;
        }
        
        return answer;
    }
}